internal/pkg/deploy/cloudformation/lb_fargate_app.go

```diff
@@ -0,0 +1,181 @@
+		return "", fmt.Errorf("execute CloudFormation template for %s: %w", c.App.Type, err)
+	}
+	return buf.String(), nil
+}
+
+// Parameters returns the list of CloudFormation parameters used by the template.
+func (c *LBFargateStackConfig) Parameters() []*cloudformation.Parameter {
+	templateParams := c.toTemplateParams()
+	return []*cloudformation.Parameter{
+		{
+			ParameterKey:   aws.String(lbFargateParamProjectNameKey),
+			ParameterValue: aws.String(templateParams.Env.Project),
+		},
+		{
+			ParameterKey:   aws.String(lbFargateParamEnvNameKey),
+			ParameterValue: aws.String(templateParams.Env.Name),
+		},
+		{
+			ParameterKey:   aws.String(lbFargateParamAppNameKey),
+			ParameterValue: aws.String(templateParams.App.Name),
+		},
+		{
+			ParameterKey:   aws.String(lbFargateParamContainerImageKey),
+			ParameterValue: aws.String(templateParams.Image.URL),
+		},
+		{
+			ParameterKey:   aws.String(lbFargateParamContainerPortKey),
+			ParameterValue: aws.String(strconv.Itoa(templateParams.Image.Port)),
+		},
+		{
+			ParameterKey:   aws.String(lbFargateRulePriorityKey),
+			ParameterValue: aws.String(strconv.Itoa(templateParams.Priority)),
+		},
+		{
+			ParameterKey:   aws.String(lbFargateRulePathKey),
+			ParameterValue: aws.String(templateParams.App.Path),
+		},
+		{
+			ParameterKey:   aws.String(lbFargateTaskCPUKey),
+			ParameterValue: aws.String(strconv.Itoa(templateParams.App.CPU)),
+		},
+		{
+			ParameterKey:   aws.String(lbFargateTaskMemoryKey),
+			ParameterValue: aws.String(strconv.Itoa(templateParams.App.Memory)),
+		},
+		{
+			ParameterKey:   aws.String(lbFargateTaskCountKey),
+			ParameterValue: aws.String(strconv.Itoa(templateParams.App.Count)),
+		},
+	}
+}
+
+// SerializedParameters returns the CloudFormation stack's parameters serialized
+// to a YAML document annotated with comments for readability to users.
+func (c *LBFargateStackConfig) SerializedParameters() (string, error) {
+	return "", nil
+}
+
+// Tags returns the list of tags to apply to the CloudFormation stack.
+func (c *LBFargateStackConfig) Tags() []*cloudformation.Tag {
+	return []*cloudformation.Tag{
+		{
+			Key:   aws.String(projectTagKey),
+			Value: aws.String(c.Env.Project),
+		},
+		{
+			Key:   aws.String(envTagKey),
+			Value: aws.String(c.Env.Name),
+		},
+		{
+			Key:   aws.String(appTagKey),
+			Value: aws.String(c.App.Name),
+		},
+	}
+}
+
+// lbFargateTemplateParams holds the data to render the CloudFormation template for an application.
+type lbFargateTemplateParams struct {
+	*deploy.CreateLBFargateAppInput
+
+	// Additional fields needed to render the CloudFormation stack.
+	Priority int
+
+	// Field types to override.
+	Image struct {
+		URL  string
+		Port int
+	}
+}
+
+func (c *LBFargateStackConfig) toTemplateParams() *lbFargateTemplateParams {
+	// TODO override app's fields with the environment's.
+
+	imgLoc := fmt.Sprintf("%s/%s/%s:%s", c.Env.Project, c.Env.Name, c.App.Name, c.ImageTag)
```

Comment:
Repository names will not include environment name.

Reply:
Yeah I thought I'd circle back and get to that after David's changes